udemy_exercises/songs/Album.java

```diff
@@ -1,9 +1,217 @@
+        return false;
+    }
+
+    public boolean addToPlaylist(String title, LinkedList <Song> playlist){
+        Song checkedSong = findSong(title);
+        if (checkedSong != null){
+            playlist.add(checkedSong);
+            return true;
+        }
+        System.out.println("The song " + title + " is not in this album.");
+        return false;
+    }
+
+    public static void main(String[] args){
+        Album album = new Album("Pilot", "Currency");
+        album.addSong("Example", 2.0);
+        album.addSong("Audio Dope 2", 3.55);
+        album.addSong("King Kong", 2.02);
+        album.addSong("Seat Change", 3.26);
+        album.addSong("Breakfast", 4.48);
+        album.addSong("Roasted", 3.17);
+        album.addSong("Skybourne", 2.51);
+        album.addSong("The Hangover", 3.22);
+        album.addSong("The Day", 3.22);
+        album.addSong("Prioritize", 3.41);
+        album.addSong("Adress", 2.41);
+        album.addSong("Life Under The Scope", 2.23);
+        albums.add(album);
+
+        album = new Album("Pilot 2", "Currency");
+        album.addSong("Airborne Aquarium", 4.11);
+        album.addSong("Michael Knight", 3.23);
+        album.addSong("Montreux", 3.39);
+        album.addSong("Famous", 4.05);
+        album.addSong("Flight Brief", 3.27);
+        album.addSong("A Gee", 3.19);
+        album.addSong("Real Estates", 3.49);
+        album.addSong("Silence", 3.25);
+        album.addSong("Hold On", 2.48);
+        album.addSong("Fashionably", 3.28);
+        album.addSong("Highed Up", 4.02);
+        album.addSong("OG", 4.18);
+        album.addSong("Michael Knight Remix", 4.36);
+        albums.add(album);
+
+        LinkedList<Song> playlist = new LinkedList<Song>();
+        albums.get(0).addToPlaylist("Example", playlist);
+        albums.get(0).addToPlaylist("Breakfast", playlist);
+
+        albums.get(0).addToPlaylist(2, playlist);
+        albums.get(1).addToPlaylist(6, playlist);
+        albums.get(1).addToPlaylist(7, playlist);
+        albums.get(1).addToPlaylist(8, playlist);
+
+        play(playlist);
+
+    }
+
+    private static void stop(){
+        System.out.println("Playlist complete.");
+    }
+
+    private static void play(LinkedList<Song> playlist){
+        Scanner scanner = new Scanner(System.in);
+        boolean quit = false;
+        boolean forward = true;
+        ListIterator<Song> listIterator = playlist.listIterator();
+        if(playlist.size() == 0){
+            System.out.println("No songs in playlist.");
+            return;
+        } else{
+            System.out.println("Npw playing " + listIterator.next().toString());
+            printMenu();
+        }
+        while(!quit){
+            int action = scanner.nextInt();
+            scanner.nextLine();
+
+            switch(action){
+                case 0:
+                    stop();
+                    quit = true;
+                    break;
+                case 1:
```

Comment:
The first time 'Play Song' is selected, it starts with the second song in the list, rather than the first.  There's a defect here.